### PR TITLE
add set_variables alias of variables<-

### LINKED ISSUE #281: [feature] set_variable_names (alias of variables<-) for piping
In many cases, one has unnamed draws, and it could be useful to have a `set_variable_names` function (equivalent to `magrittr::set_col_names`) which could be used in piping workflows. This could be particularly useful when using prediction functions that return unnamed draws.

Functionally this would work like `variables<-` but would allow for piping.

For example
```{r}
unnamed_draws|> 
    set_variable_names(c("m1", "m2")) # here we set all variable names at once
```
equivalent to
```
variables(unnamed_draws) <- c("m1", "m2")
```

### PR DESCRIPTION
**Pull request by n-kall**
#### Summary

Adds `set_variables` function which is an alias of `variables<-`. Fixes #281

Example functionality:

```{r}
x <- as_draws(matrix(rnorm(100), ncol = 2))
variables(x)
#> [1] "...1" "...2"

x |> set_variables(c("theta[1]", "theta[2]")) |> variables()
#> [1] "theta[1]" "theta[2]"

# this is equivalent to
variables(x) <- c("theta[1]", "theta[2]")
variables(x)
#> [1] "theta[1]" "theta[2]"

```

#### Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by paul-buerkner** [comment]
Thank you! Would you mind moving the code to the R file where variables() <- is defined to keep them together? Same with the tests.

**Comment by n-kall** [comment]
``

> Would you mind moving the code to the R file where variables() <- is defined to keep them together? Same with the tests.

Done!

File: R/draws-index.R
Comment (paul-buerkner) [inline R/draws-index.R:161]: Please avoid the use of the pipe in the docs because this may lead to a failur in old R versions not having the pipe.

File: R/draws-index.R
Comment (paul-buerkner) [inline R/draws-index.R:168]: why use .x as the first argument here? Wouldn't x be okay too?

Comment (paul-buerkner) [inline R/draws-index.R:168]: shall the generic enforce the second argument be called "variables"? right now the generic doesn't specify that so other methods could in theory call the argument differently which might end up being confusing, I think.

**Review by paul-buerkner — CHANGES_REQUESTED** [review]
Thank you! I have a few minor comments before merging this.

File: R/draws-index.R
Comment (n-kall) [reply R/draws-index.R]: Fixed by changing variable to `x`

Comment (n-kall) [reply R/draws-index.R]: This was an oversight, fixed now so that the generic has a variables argument

File: R/draws-index.R
Comment (n-kall) [reply R/draws-index.R]: Fixed

File: R/draws-index.R
Comment (mjskay) [inline R/draws-index.R:175]: Why make `set_variables` generic? If it is a non-generic function that just forwards to `variables<-` then implementers of other classes need only override the generic there.

Comment (n-kall) [reply R/draws-index.R:175]: I didn't think too much about this either way, as I just used `rename_variables` as a template, which is a generic with only a `rename_variables.draws` method defined. If it's better to change it to a non-generic function, I can do that

Comment (mjskay) [reply R/draws-index.R:175]: I defer to @paul-buerkner, but IMO it would be simpler :).

Comment (jgabry) [reply R/draws-index.R:175]: non-generic sounds good to me too

Comment (n-kall) [reply R/draws-index.R]: I changed it to non-generic now

File: R/draws-index.R
Comment (paul-buerkner) [inline R/draws-index.R:161]: you are not assigning to x here. Is that intentional?

**Review by paul-buerkner — CHANGES_REQUESTED** [review]
looks good. One minor thing before I will merge

File: R/draws-index.R
Comment (n-kall) [reply R/draws-index.R]: Good catch! Fixed this now.

**Review by paul-buerkner — APPROVED** [review]

**Comment by paul-buerkner** [comment]
Thank you for adding this!

**Comment by n-kall** [comment]
> Thank you for adding this!

Thanks for the review!

**Pull request merged**
